src/core/batch.rkt

```diff
@@ -1,15 +1,19 @@
 #lang racket
 
 (require "../utils/timeline.rkt"
-         "../syntax/syntax.rkt")
+         "../syntax/syntax.rkt"
+         "../syntax/types.rkt")
 
-(provide progs->batch
-         batch->progs
+(provide progs->batch ; (Listof Expr) -> Batch
+         batch->progs ; Batch -> (Listof Expr)
          (struct-out batch)
-         batch-length
-         batch-ref
-         deref
-         batch-replace)
+         (struct-out batchref) ; temporarily for patch.rkt
+         batch-length ; Batch -> Integer
+         batch-ref ; Batch -> Index -> Expr
```

Comment:
Here by `Expr` you mean `Batchref` right?

Reply:
In this version `batch-ref` indeed returns `Expr`, basically extracts expression from the nodes
I am creating `Batchref` by using structure `batchref` where it is needed.
We probably need to rename these functions to avoid misunderstandings